setup.py

```diff
@@ -115,13 +115,21 @@ class PkgTest(test):
     for the "test" command.
     """
 
-    new_commands = [('test_ext', lambda self: True), ('test_doc', lambda self: True)]
+    user_options = test.user_options + [
+        ('no-doc-test', None, "Do not test documentation snippets"),
+    ]
+    new_commands = [('test_ext', lambda self: true), ('test_doc', lambda self: not self.no_doc_test)]
     sub_commands = test.sub_commands + new_commands
 
+    def initialize_options(self):
+        super().initialize_options()
+        self.no_doc_test = False
+
     def run(self):
         super().run()
         self.run_command('test_ext')
-        self.run_command('test_doc')
+        if not self.no_doc_test:
+            self.run_command('test_doc')
```

Comment:
Sorry seeing this so late.
Since `test_dec` is a sub_command with a predicate that will return False in case `self.no_doc_test` is set, I would expect `run_command` to skip executing it (so maybe we can drop the if here?)

Reply:
Indeed, according to my understanding of how it works, the test should not be required. But if I remember correctly, the test was still required...

Did you test your suggestion?